src/txn/txn_ckpt.c

```diff
@@ -237,7 +237,6 @@ __checkpoint_data_source(WT_SESSION_IMPL *session, const char *cfg[])
 int
 __wt_checkpoint_list(WT_SESSION_IMPL *session, const char *cfg[])
 {
-	WT_DATA_HANDLE *saved_dhandle;
```

Comment:
That's exactly right: I have tried to flip things so the caller is responsible for the save/restore.  In this cases there is only one (direct) caller, which now does the save/restore.